src/Redux/Books/Books.js

```diff
@@ -1,23 +1,27 @@
 const ADD_BOOK = 'bookstore/books/addBook';
 const REMOVE_BOOK = 'bookstore/books/removeBook';
-const defaultData = [
-  { id: 1, title: 'A book to you', author: 'Esmatullah' },
-  { id: 2, title: 'A book of better life', author: 'Ahmadzai' },
-];
-export default function removeReducer(state = defaultData, action) {
+const FETCH_DATA = 'bookstore/books/fetchData';
+export default function addRemoveReducer(state = [], action) {
   switch (action.type) {
+    case FETCH_DATA:
+      return [...action.payload];
     case ADD_BOOK:
-      return [...state, action.books];
+      return [...state, action.payload];
     case REMOVE_BOOK:
       return state.filter((book) => book.id !== action.id);
     default:
       return state;
   }
 }
 
+export const fetchData = (books) => ({
+  type: FETCH_DATA,
+  payload: books,
+});
+
 export const addBook = (book) => ({
   type: ADD_BOOK,
-  books: book,
+  payload: book,
 });
 
 export const removeBook = (id) => ({
```

Comment:
Using `createAsyncThunk` is a must for this project. Kindly use the same for making asynchronous calls to the API. Also, all the API calls should be made from the redux instead of the components. Remember to keep the functional components side effects free.